widget/ui/src/components/Modal/Modal.tsx

```diff
@@ -91,22 +91,27 @@ export function Modal(props: PropsWithChildren<ModalPropTypes>) {
               onTransitionEnd={handleTransitionEnd}
               ref={ref}>
               {header ?? (
-                <ModalHeader noTitle={!title && dismissible && !prefix}>
-                  {prefix}
-                  {title && (
+                <Header
+                  prefix={prefix}
+                  title={
                     <Typography variant="title" size="small">
                       {title}
                     </Typography>
-                  )}
-                  <Flex>
-                    {suffix}
-                    {dismissible && hasCloseIcon && (
-                      <IconButton onClick={onClose} variant="ghost">
-                        <CloseIcon color="gray" size={14} />
-                      </IconButton>
-                    )}
-                  </Flex>
-                </ModalHeader>
+                  }
+                  suffix={
+                    <Flex>
+                      {suffix}
+                      {dismissible && hasCloseIcon && (
+                        <IconButton onClick={onClose} variant="ghost">
+                          <CloseIcon color="gray" size={14} />
+                        </IconButton>
+                      )}
+                    </Flex>
+                  }
+                  transparent
+                  css={{ padding: '$20 $20 $0 $20', minHeight: '44px' }}
```

Comment:
Generally, you shouldn't need this. I didn't run code yet, I will take a look at your requirement here after fixing my other comments.

Reply:
the padding of header of modal is different from header of pages (layout) so I customized it here.

Reply:
Let's keeping Header out of Modal please. maybe it's not related to your changes directly but it drives me crazy.
At least we can change it to:
```
{props.header}
<Content> ....
```

A better improvement can be completely remove `header` prop from modal. Modal should have a single responsibily, not layouting or other stuff. when someone is going to use `Modal` it would be like:

```
<Modal>
<Header />
<WhateverContent />
</Modal>
```

Reply:
All improvements related to Modal component will be added in another PRs.